misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -148,7 +149,22 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
             if (BEFORE_SUITE_ANNOTATION_NAME.equals(annotationName)) {
                 suite.addBeforeSuiteFunction(functionName);
             } else if (AFTER_SUITE_ANNOTATION_NAME.equals(annotationName)) {
-                suite.addAfterSuiteFunction(functionName);
+                AtomicBoolean alwaysRun = new AtomicBoolean(false);
+                if (attachmentNode.getExpression() instanceof BLangRecordLiteral) {
```

Comment:
Instead of `instanceof` can we check the node kind? i.e. `attachmentNode.getExpression().getKind() == NodeKind.RECORD_LITERAL_EXP`

Reply:
Fixed in f9d41bc